File: 13-04-2024/Reverse Bits.py
#User function Template for python3

class Solution:
    def reversedBits(self, x):
        #Your Code Start
        ans = 0
        i = 0
        
        while x != 0:
            bit = x & 1
            ans |= bit << (31 - i)
            x >>= 1
            i += 1
        
        # If the original number was negative, set the leftmost bit to 1
        if ans < 0:
            ans = 4294967296 + ans  # 4294967296 = 2^32
        
        return ans
        #Your Code End
        


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        X=int(input())
        
        ob = Solution()
        print(ob.reversedBits(X))
# } Driver Code Ends

File: 13-04-2024/Reverse Bits.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            Long X = Long.parseLong(read.readLine());

            Solution ob = new Solution();
            System.out.println(ob.reversedBits(X));
        }
    }
}
// } Driver Code Ends


// User function Template for Java

class Solution {
    static Long reversedBits(Long x) {
        //Your Code Start
         long ans = 0;
        long i = 0;
        
        while (x != 0) {
            long bit = x & 1;
            ans |= bit << (31 - i);
            x >>= 1;
            i++;
        }
        
        // If the original number was negative, set the leftmost bit to 1
        if (ans < 0) {
            ans = (long) (Math.pow(2, 32) + ans); // 2^32
        }
        
        return ans;
        //Your Code End
    }
};
